### DISCUSSION TITLE
Change root folder ~/.config/nvim when use Lazy update

### DISCUSSION THREAD

**Discussion by nguyen-quang-phu** [LazyVim/LazyVim#3951] [Q&A]
Hi all, recently I move to use nix to manage my dotfiles, and i use nix to generate nvim config with folder "readonly", when i use lazy update, it can not change lazyvim.json and lazy-lock.json, can i custom root nvim config by another path diff "~/.config/nvim". This is my manage nix repo https://github.com/nguyen-quang-phu/nix-darwin
I setting nvim config like this: 
![image](https://github.com/LazyVim/LazyVim/assets/43996827/d8abe91c-28fe-4785-8d1d-b4de98bfb331)
It's simly i copy from ~/.config/nvim to nix-darwin and after nix will generate my config. Very happy if someone gives me a solution.

**Comment by dpetka2001** [comment]
`lazy-lock.json`'s location can be changed from `lazy.nvim`'s settings. Take a look at the [configuration](https://lazy.folke.io/configuration), you're looking for the `lockfile` setting just change that path to what you want. 

For `lazyvim.json` you can put in your `init.lua` file before the line with `require("config.lazy")` the following `vim.g.lazyvim_json = "whatever_path_you_want"` and that should also change the path.

**Reply by nguyen-quang-phu** [reply]
thank you, vim.g.lazyvim_json seem not work for me, but I found another solution with lazyvim_json , lazy-lock.json seem look good.
![image](https://github.com/LazyVim/LazyVim/assets/43996827/12873ef8-31f6-4ded-b631-6adf327dc9ef)

**Reply by dpetka2001** [reply]
Strange. In `~/.config/nvim/init.lua` I have the following 
```lua
vim.g.lazyvim_json = "/home/user_name/Desktop/lazyvim.json`
require("config.lazy")
```
and it seems to work for me. It creates the `lazyvim.json` file under `~/Desktop` directory. But if you found another solution that works for you then all good.

**Reply by lbesnard** [reply]
That worked fine for me, thanks. In case someone is interested, here is my lazyvim setup on nixos
https://github.com/lbesnard/nixos_config/blob/main/config/neovim.nix
